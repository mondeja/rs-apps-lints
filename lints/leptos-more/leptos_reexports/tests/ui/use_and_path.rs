pub mod leptos {
    pub mod serde {
        pub struct Serialize;
    }

    pub mod wasm_bindgen {
        pub struct JsCast;
    }

    pub mod web_sys {
        pub struct Element;
        pub struct HtmlElement;
    }

    pub mod prelude {
        pub struct Signal;
    }

    pub mod tracing {
        pub struct Instrument;
    }

    pub mod serde_json {
        pub struct Value;
    }
}

fn r#use() {
    // single
    use leptos::wasm_bindgen::JsCast;
    let _ = JsCast;

    // root
    use ::leptos::serde::Serialize;
    let _ = Serialize;

    // list stem
    #[allow(unused_imports)]
    use leptos::{wasm_bindgen, serde};
    #[allow(unused_imports)]
    use {leptos::serde_json, leptos::tracing};

    // complex
    use leptos::{prelude::*, web_sys::HtmlElement};
    let _ = HtmlElement;
    let _ = Signal;

    // glob
    use leptos::web_sys::*;
    let _ = Element;

    // wildcard
    #[allow(unused_imports)]
    use leptos::*;

    // wildcard root
    #[allow(unused_imports)]
    use ::leptos::*;
}

fn inline() {
    // single
    let _ = leptos::wasm_bindgen::JsCast;

    // root
    let _ = ::leptos::serde::Serialize;
}

fn main() {
    r#use();
    inline();
}
