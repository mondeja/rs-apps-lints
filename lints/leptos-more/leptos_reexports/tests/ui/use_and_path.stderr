warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:30:17
   |
LL |     use leptos::wasm_bindgen::JsCast;
   |                 ^^^^^^^^^^^^
   |
   = help: consider using `use wasm_bindgen::JsCast` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme
   = note: `#[warn(leptos_reexports)]` on by default

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:34:19
   |
LL |     use ::leptos::serde::Serialize;
   |                   ^^^^^
   |
   = help: consider using `use leptos::serde::Serialize` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:39:18
   |
LL |     use leptos::{wasm_bindgen, serde};
   |                  ^^^^^^^^^^^^
   |
   = help: consider using `use wasm_bindgen` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:39:32
   |
LL |     use leptos::{wasm_bindgen, serde};
   |                                ^^^^^
   |
   = help: consider using `use serde` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:41:18
   |
LL |     use {leptos::serde_json, leptos::tracing};
   |                  ^^^^^^^^^^
   |
   = help: consider using `use serde_json` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:41:38
   |
LL |     use {leptos::serde_json, leptos::tracing};
   |                                      ^^^^^^^
   |
   = help: consider using `use tracing` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:44:30
   |
LL |     use leptos::{prelude::*, web_sys::HtmlElement};
   |                              ^^^^^^^
   |
   = help: consider using `use web_sys::HtmlElement` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:49:17
   |
LL |     use leptos::web_sys::*;
   |                 ^^^^^^^
   |
   = help: consider using `use web_sys::*` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:54:5
   |
LL |     use leptos::*;
   |     ^^^^^^^^^^^^^^
   |
   = help: do not import using `leptos::*`. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:58:5
   |
LL |     use ::leptos::*;
   |     ^^^^^^^^^^^^^^^^
   |
   = help: do not import using `leptos::*`. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:63:21
   |
LL |     let _ = leptos::wasm_bindgen::JsCast;
   |                     ^^^^^^^^^^^^
   |
   = help: consider using `wasm_bindgen::JsCast` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: usage of a third party library re-export from `leptos`
  --> $DIR/use_and_path.rs:66:23
   |
LL |     let _ = ::leptos::serde::Serialize;
   |                       ^^^^^
   |
   = help: consider using `leptos::serde::Serialize` instead. For further information visit https://github.com/mondeja/rs-apps-lints/tree/main/lints/leptos-more/leptos_reexports#readme

warning: 12 warnings emitted

